src/FuncSharp/Option/Option.cs

```diff
@@ -64,88 +65,144 @@ public static IOption<A> Empty<A>()
         }
     }
 
-    internal class Option<A> : Coproduct2<A, Unit>, IOption<A>
+    internal sealed class Option<A> : IOption<A>
     {
-        static Option()
```

Comment:
Static constructor removes `beforefieldinit` flag from class definition, which leads to huge performance penalty.